iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkIndex.java

```diff
@@ -50,14 +53,53 @@ void checkAndAdd(Identifiable<?> obj) {
         all.add(obj);
     }
 
+    void addAlias(Identifiable<?> obj, String alias) {
+        if (objectsById.containsKey(alias)) {
+            Identifiable<?> aliasConflict = objectsById.get(alias);
+            String message = String.format("Object (%s) with alias '%s' cannot be created because alias already refers to object (%s) with ID '%s'",
+                    obj.getClass(),
+                    alias,
+                    aliasConflict.getClass(),
+                    aliasConflict.getId());
+            throw new PowsyblException(message);
+        } else if (idByAlias.containsKey(alias)) {
+            Identifiable<?> aliasConflict = objectsById.get(idByAlias.get(alias));
+            String message = String.format("Object (%s) with alias '%s' cannot be created because alias already refers to object (%s) with ID '%s'",
+                    obj.getClass(),
+                    alias,
+                    aliasConflict.getClass(),
+                    aliasConflict.getId());
+            throw new PowsyblException(message);
+        }
+        idByAlias.put(alias, obj.getId());
```

Comment:
You should test if the object that already exists in the map is not the same than the `obj` argument, and silently ignore it. You could be possible to write something like this:
```
addAlias(obj, "alias"); // works because "alias" is not used
addAlias(obj, "alias"); // works because "alias" is already an alias of obj
addAlias(obj2, "alias"); // fails because "alias" is already an alias of obj, and obj != obj2
```